app/Repositories/Doctrine/AccountRepository.php

```diff
@@ -0,0 +1,42 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Repositories\Doctrine;
+
+use App\Database\Entities\Account;
+use App\Database\Exceptions\InvalidSubscriptionTypeException;
+use App\Repositories\Doctrine\ORM\AbstractRepository;
+use App\Repositories\Interfaces\AccountRepositoryInterface;
+
+final class AccountRepository extends AbstractRepository implements AccountRepositoryInterface
+{
+    /**
+     * Find by subscription type.
+     *
+     * @param string $subscriptionType
+     *
+     * @return mixed[]
+     */
+    public function findBySubscriptionType(string $subscriptionType): array
+    {
+        try {
```

Comment:
We may get rid of try and catch here, just allow it to fail :) if something happens we want to know what error it is